spec/app/github_repo_fetcher_spec.rb

```diff
@@ -17,27 +17,36 @@ def stub_cache
     cache
   end
 
+  describe "#instance" do
+    it "acts as a singleton" do
+      expect(GitHubRepoFetcher.instance).to be_a_kind_of(GitHubRepoFetcher)
+      expect(GitHubRepoFetcher.instance).to eq(GitHubRepoFetcher.instance)
+    end
+  end
+
   describe "#repo" do
     it "fetches a repo from cache if it exists" do
       allow(stub_cache).to receive(:fetch).with("all-repos", hash_including(:expires_in)) do
-        [public_repo]
+        some_repo = public_repo
+        allow(some_repo).to receive(:name).and_return("some-repo")
```

Comment:
This addition was needed as otherwise we would intermittently see:

```
  1) GitHubRepoFetcher#repo fetches a repo from cache if it exists
     Failure/Error: all_alphagov_repos.find { |repo| repo.name == repo_name } || raise("alphagov/#{repo_name} not found")
       #<Double "Public repo"> received unexpected message :name with (no args)
     # ./app/github_repo_fetcher.rb:10:in `block in repo'
     # ./app/github_repo_fetcher.rb:10:in `each'
     # ./app/github_repo_fetcher.rb:10:in `find'
     # ./app/github_repo_fetcher.rb:10:in `repo'
     # ./spec/app/github_repo_fetcher_spec.rb:33:in `block (3 levels) in <top (required)>'
```